test/suite/test_layered32.py

```diff
@@ -0,0 +1,239 @@
+        else:
+            self.assertEqual(self.get_stat(stat.conn.cache_read_internal_delta), 0)
+
+        follower_config = self.conn_base_config + 'disaggregated=(role="follower"),'
+        self.reopen_disagg_conn(follower_config)
+        time.sleep(1.0)
+
+        # Verify the updated values in the table.
+        self.verify(kv_modfied, inital_value)
+
+        # Assert that we have constructed at least one internal page delta.
+        if (self.delta_type == 'both' or self.delta_type == 'internal_only'):
+            self.assertGreater(self.get_stat(stat.conn.cache_read_internal_delta), 0)
+        else:
+            self.assertEqual(self.get_stat(stat.conn.cache_read_internal_delta), 0)
+
+    def test_internal_page_delta_random(self):
+
+        self.session.create(self.uri, self.session_create_config())
+
+        # Populate the table with nitems.
+        inital_value = "abc" * 100
+        small_value = "b123r" * 5
+        inital_ts = 5
+
+        # Initial Population
+        kv = {str(i): small_value for i in range(1, self.nitems + 1)}
+        self.insert(kv, inital_ts)
+    
+        # First Checkpoint & Reopen
+        self.session.checkpoint()
+        self.reopen_disagg_conn(self.conn_config())
+
+        # Large Value Insertion
+        keys_to_update = ["4041","4042","4043", "4044", "4045", "4046", "4047", "4048", "4049", "4050"]
+        kv_modified = {}
+        for key in keys_to_update:
+            kv = {key: inital_value}
+            inital_ts = inital_ts + 1
+            self.insert(kv, inital_ts)
+
+        #Second Checkpoint 
+        self.session.checkpoint()
+
+        # Small Value Update
+        for key in keys_to_update:
+            kv = {key: small_value}
+            inital_ts = inital_ts + 1
+            self.insert(kv, inital_ts)
+            self.session.checkpoint()
+            kv_modified.update(kv)
+
+        # Assert that we have written at least one internal page delta.
+        if (self.delta_type == 'both' or self.delta_type == 'leaf_only'):
+            self.assertGreater(self.get_stat(stat.conn.rec_page_delta_leaf), 0)
+        if (self.delta_type == 'both' or self.delta_type == 'internal_only'):
+            self.assertGreater(self.get_stat(stat.conn.rec_page_delta_internal), 0)
+        if (self.delta_type == 'none'):
+            self.assertEqual(self.get_stat(stat.conn.rec_page_delta_leaf), 0)
+            self.assertEqual(self.get_stat(stat.conn.rec_page_delta_internal), 0)
+
+        # Verify the updated values in the table.
+        self.verify(kv_modified, small_value)
+
+        # Re-open the connection to clear contents out of memory.
+        self.reopen_disagg_conn(self.conn_config())
+
+        # Verify the updated values in the table.
+        self.verify(kv_modified, small_value)
+
+        # Re-open the connection to clear contents out of memory.
+        self.reopen_disagg_conn(self.conn_config())
```

Comment:
We reopen but we do nothing?

Reply:
Removed redundant code in 24de87e2d6d1b1389c833c2ee1c6b5455659197c